From 6cafae82f5014fd820a6871a0855052e4b6de17f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?L=C3=A1szl=C3=B3=20Vask=C3=B3?=
 <1771332+vlaci@users.noreply.github.com>
Date: Fri, 1 Mar 2024 17:09:36 +0100
Subject: [PATCH] fetchers: always use git work dir state for local
 repositories

This makes secrets decrypted by git hooks available
---
 src/libfetchers/git.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libfetchers/git.cc b/src/libfetchers/git.cc
index f8d89ab2f..7262a9747 100644
--- a/src/libfetchers/git.cc
+++ b/src/libfetchers/git.cc
@@ -452,7 +452,7 @@ struct GitInputScheme : InputScheme
            allow fetching directly from a dirty workdir. */
         if (!input.getRef() && !input.getRev() && isLocal) {
             auto workdirInfo = getWorkdirInfo(input, actualUrl);
-            if (!workdirInfo.clean) {
+            if (!workdirInfo.clean || true) {
                 return fetchFromWorkdir(store, input, actualUrl, workdirInfo);
             }
         }
-- 
2.43.1

